src/Datadog.Trace/Abstractions/ISpan.cs

```diff
@@ -9,9 +9,9 @@ internal interface ISpan
         string Type { get; set; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether this span represents an error.
+        /// Gets or sets the span's execution status
         /// </summary>
-        bool Error { get; set; }
+        SpanStatus Status { get; set; }
```

Comment:
Datadog won't be able to accept this change...even though it is an internal interface, this may cause a crash in Datadog automatic instrumentation. I can speak more to it, but we can't accept this. Perhaps we can add a new interface member instead?

Reply:
Can you please give repro steps and/or what is going to cause the issue? It will help me avoid similar issues in future.

Reply:
The steps for this situation are:
1. .NET Core application
2. Datadog automatic instrumentation enabled for the application, using a lower version (say 1.24.0)
3. Datadog NuGet package added to the application, using a higher version that has an internal API removed

Result: `MissingMethodException`. This occurs because the automatic instrumentation (`Datadog.Trace.ClrProfiler.Managed`) has an assembly dependency on the manual instrumentation (`Datadog.Trace`) and is allowed to make internal calls across assemblies. However, using a NuGet package allows the application developer to add a newer version than what is expected and the type loading will unify on the newer version, so issues occur when the API exists on the older `Datadog.Trace` but are removed in a later `Datadog.Trace`. We are working to resolve this issue but we have not completed that work yet.

Reply:
Can you just confirm that it would be reproducible only for https://github.com/DataDog/dd-trace-go if you just merge these changes as it is? To reproduce it here I would need to create NuGet package for OTel .NET AutoInstrumentation manually.

### Option 1
Make the internal API backwards compatible.

### Option 2
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation does not support Datadog NuGet package. It should support only the OpenTracing NuGet package at the moment. In future, it could support .NET Activity API and OpenTelemetry SDK.
Datadog can always make add the "backward compatibility" when merging this PR to the https://github.com/DataDog/dd-trace-dotnet repository.

I just wanted to propose **Option 2** as I see more and more problems connected with "API compatibility" (e.g. here https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/72). I do not want to force any solution. I just think we should think about it more and maybe even describe the options with pros and cons in some Google Docs and discuss it during the next SIG meeting(s).

Reply:
> Datadog can always make add the "backward compatibility" when merging this PR to the https://github.com/DataDog/dd-trace-dotnet repository.

Yeah I am on board with this change, we can make the API compatibility change in the dd-trace-dotnet repo. I think Paulo's comment about adding a mechanism to detect API changes would be really helpful to identify changes like this in the future.

Reply:
Is there an Option 3, which I think is out of scope for this PR, where we start to build a new OTel specific tracer API implementation? This tracer would only need to implement an API surface area that OTel needs, and would allow the APIs to operate independently. We are starting to see more and more points of friction to maintain compatibility, and we may need to start carefully choosing where and where not to maintain compatibility to allow progress to continue.

Reply:
@zacharycmontoya That is very kind. Even though I do not want to rush. I feel this is an important design decision that I that SIG would have a "quorum" agreement. Maybe there are some other ways to make life easier for all of us. Maybe we could make a brainstorming session during SIG meeting 😉 

@nrcventura I was thinking of **Option 3** even when I was analyzing the semantic conventions for attributes/tags. I felt the number of changes would be unrealistic but now I think it is worth considering it again. It is good to have as many options as possible 👍 I will edit my comment and add your option proposal.